lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -174,6 +174,11 @@ def add_resource_relationships(attrs, serializer, options = {})
               end
             end
 
+            if association.respond_to?(:json_api_links) &&
+               @options[:include].include?(name.to_s)
+              add_links(attrs[:relationships][name], association)
```

Comment:
That's right, so we have to differentiate between "relationship_links" and "resource_links" ?! 

then i would rename "json_api_links" to "resource_links"
and use the new one: "relationship_links" for this ?